storage/client_test.go

```diff
@@ -57,21 +57,20 @@ func createTestStoreWithEngine(t *testing.T, eng engine.Engine, clock *hlc.Clock
 	bootstrap bool, context *storage.StoreContext) (*storage.Store, *util.Stopper) {
 	stopper := util.NewStopper()
 	rpcContext := rpc.NewContext(hlc.NewClock(hlc.UnixNano), security.LoadInsecureTLSConfig(), stopper)
-	var ctx *storage.StoreContext
 	if context == nil {
-		ctx = &storage.TestStoreContext
-	} else {
-		ctx = context
+		// make a copy
+		ctx := storage.TestStoreContext
+		context = &ctx
```

Comment:
Maybe it's premature to treat go-nyet as mandatory. The ratio of code churn to net bugs fixed/prevented isn't good. We've had more problems in practice with copying vs non-copying than we have with shadowing variables, so we shouldn't sacrifice clarity in this area.